cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -2182,5 +2185,146 @@ TEST(TestRankForFixedSizeBinary, RankFixedSizeBinary) {
       ArrayFromJSON(binary_type, R"(["aaa", "   ", "eee", null, "eee", null, "   "])"));
 }
 
+TEST(TestRankForReal, RankRealChunked) {
+  for (auto real_type : ::arrow::FloatingPointTypes()) {
+    for (auto null_placement : AllNullPlacements()) {
+      for (auto tiebreaker : AllTiebreakers()) {
+        for (auto order : AllOrders()) {
+          AssertRankEmpty(real_type, order, null_placement, tiebreaker);
+        }
+
+        AssertRankSimple(
+            ChunkedArrayFromJSON(real_type, {"[2.1, 3.2]", "[1.0, 0.0, 5.5]"}),
+            null_placement, tiebreaker);
+      }
+    }
+    AssertRankAllTiebreakers(
+        ChunkedArrayFromJSON(real_type, {"[1.2, 0.0]", "[5.3, null, 5.3, null, 0.0]"}));
+  }
+}
+
+TEST(TestRankForIntegral, RankIntegralChunked) {
```

Comment:
Can you add some tests for zero chunks and/or empty chunks?

Reply:
I think `AssertRankEmpty` already does this but let me know if you had something else in mind